server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -2007,8 +2007,16 @@ sub int as round((random() + 1) * 100)
 
         execute("INSERT INTO tbl(a) VALUES (1)");
         refresh();
+
         execute("SELECT o1['sub'] FROM tbl WHERE a = 1");
-        assertThat((int) response.rows()[0][0]).isGreaterThan(0);
+        int generated = (int) response.rows()[0][0];
+        assertThat(generated).isGreaterThan(0);
+
+        // some iterations to ensure it hits both primary and replica
+        for (int i = 0; i < 30; i++) {
```

Comment:
I think it's too much, maybe 10 is enough,
but also maybe consider using the `public SQLResponse execute(String stmt, Object[] args, String node)` variant where you can pass the node.

Reply:
Thanks, updated the test. Took a slightly different approach to ensure running query on 2 data nodes.

Reply:
thx!

Reply:
had to get back repeated select. But I confirmed that 10 is enough. See https://github.com/crate/crate/pull/15272